pyocd/debug/elf/decoder.py

```diff
@@ -184,7 +184,10 @@ def _build_line_search_tree(self):
                 if prevstate and not skipThisSequence:
                     fileinfo = lineprog['file_entry'][prevstate.file - 1]
                     filename = fileinfo.name
-                    dirname = lineprog['include_directory'][fileinfo.dir_index - 1]
+                    try:
+                        dirname = lineprog['include_directory'][fileinfo.dir_index - 1]
+                    except IndexError:
+                        dirname = '<unknown dir>'
```

Comment:
Can you change the unknown dirname to just "<unknown>"? We already know it's a dir because of the attribute to which it is attached.

Reply:
I actually meant "\<unknown\>", but Markdown apparently thought that was an HTML tag. Oops. Anyway, I'm perfectly fine with empty strings!